Editor/MenuActions/Editors/NewShapeToggle.cs

```diff
@@ -11,7 +11,7 @@
 
 namespace UnityEditor.ProBuilder.Actions
 {
-    sealed class NewShapeToggle : MenuToolToggle
+    sealed class NewShapeToggle : MenuAction
```

Comment:
Great improvement on that.
Regarding the MenuToolToggle remark though regarding the fact that this PR removes public API by removing MenuToolToggle class.